### PR TITLE
Ora2 events

### PR DESCRIPTION
**Pull request by quitterie-lcs**
# Purpose

ORA2 events describe the activity for assessments pair reviewing.
Edx defined models and their selectors are described with Pydantic.

# Proposal

Define Pydantic model for: 

- openassessmentblock.create_submission,
- openassessmentblock.get_peer_submission,
- openassessmentblock.get_submission_for_staff_grading,
- openassessmentblock.peer_assess,
- openassessmentblock.save_submission,
- openassessmentblock.self_assess,
- openassessmentblock.staff_assess,
- openassessmentblock.student_training_assess_example,
- openassessment.submit_feedback_on_assessments,
- openassessment.upload_file,

### CODE REVIEW

**Review by SergioSim — COMMENTED** [review]
This looks awesome! :muscle: 
Have only some minor questions & suggestions)
Also, [here](https://pastebin.com/6UPBRAwX) are some ora2 events generated using the learning-analytics-playground to battle-test the models) Sorry for the long delay(

image

File: src/ralph/models/edx/open_response_assessment/fields/events.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        scorer_id (str): Consists of the course-specific anonymized user ID of the
+            learner who submitted the assessment.
+        score_type (str): Consists of either `PE` value for a peer assessment, `SE` for
+            a self assessment or `ST` for a staff assessment.
+        submission_uuid (str): Consists of the unique identifier for the submitted
+            response.
+    """
+
+    feedback: constr(max_length=10000)
+    parts: list[dict]
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/ralph/models/edx/open_response_assessment/fields/events.py:96]: That sounds better

File: src/ralph/models/edx/open_response_assessment/fields/events.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+    Attributes:
+        text (str): Consists of the answer field.
+        file_upload_key (str): Consists of AWS S3 key identifying the location of the
+            uploaded file on the Amazon S3 storage service. Only used when an answer
+            includes files.
+    """
+
+    text: str
+    file_upload_key: str
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/ralph/models/edx/open_response_assessment/fields/events.py:145]: It differs from the documentation and these fields are the ones contained in `answer` ?

[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/ralph/models/edx/open_response_assessment/fields/events.py]: It does not seem to be there anymore when looking [submission_mixin.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/xblock/submission_mixin.py#L381-L411) from latest version of OpenEdx repo

**Review by wilbrdt — COMMENTED** [review]
Alt text
Looks good! Just some minor questions/suggestions

File: src/ralph/models/edx/open_response_assessment/fields/events.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            the course.
+        submission_returned_uuid (str): Consists of the unique identifier of the
+            response that was retrieved for assessment. Set to `None` if no assessment
+            available.
+        requesting_staff_id (str): Consists of the course-specific anonymized user ID
+            of the course team member who is retrieved the response for grading.
+        type (str): Consists of the type of staff grading that is being performed.
+            Currently set to `full-grade`.
+    """
+
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/ralph/models/edx/open_response_assessment/fields/events.py:59]: A comment from @SergioSim: https://github.com/openfun/ralph/pull/226#discussion_r1044392026
It has been noticed that the documentation was not entirely correct regarding this statement

File: src/ralph/models/edx/open_response_assessment/fields/events.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        parts (dict): Consists of a key-value dictionary with all answers text.
+        file_keys (list): Consists of a list of file identifiers if files are given for
+            answer.
+        files_description (list): Consists of a list of file descriptions if files are
+            given for answer.
+    """
+
+    parts: List[Dict[Literal["text"], str]]
+    file_keys: Optional[List[str]]
+    files_descriptions: Optional[List[str]]
```
[the remark this answers is not in the document]
Comment (quitterie-lcs) [reply src/ralph/models/edx/open_response_assessment/fields/events.py:191]: Yes, see [create_submission](https://github.com/openfun/edx-ora2/blob/559fae10e84e2bacad5548950d6d7adcb29da409/openassessment/xblock/submission_mixin.py#L228) method

[the remark this answers is not in the document]
Comment (wilbrdt) [reply src/ralph/models/edx/open_response_assessment/fields/events.py:191]: Nice catch, thanks!

**Review by wilbrdt — APPROVED** [review]
Alt text

**Pull request merged**